### ISSUE TITLE
Deploy failure with Invalid character in header content ["Authorization"] - cannot troubleshoot due to missing error info

### ISSUE THREAD

**Issue by alexanderxc** [serverless/serverless#11207]
### Are you certain it's a bug?

- [x] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

**Context:** Alexa Skill built in jovo

**Behavior**
When trying to deploy the bundled code via serverless to AWS lambda the command is unsuccessful, failing with error: 

```bash
Invalid character in header content ["Authorization"]
```

Unfortunately, the `--verbose` option does not pride additional information.

The issue does not appear to be linked to Jovo as the bundle occurs correctly. The issue occurs when trying to deploy to AWS and everything gets stuck at the CloudFormation retrieve step. 

As well, even using AWS credentials with unrestricted permission do not appear to allow deploy. 

I've searched for similar issue but could not find anything relevant. Please assist. 

Is there a way to troubleshoot further and see what is happening under the hood?

### Service configuration (serverless.yml) content

```yaml
"service": "x-voice"
"frameworkVersion": "3"
"package":
  "artifact": "./bundle.zip"
"provider":
  "name": "aws"
  "runtime": "nodejs14.x"
  "region": "eu-central-1"
  "stage": "prod"
"functions":
  "handler":
    "handler": "index.handler"
    "url": true
    "timeout": 7
    "events":
      - "alexaSkill": "amzn1.ask.skill.some-value"
```

### Command name and used flags

serverless-deploy --verbose

### Command output

```shell
Deploying x-voice to stage prod (eu-central-1)

Packaging
Retrieving CloudFormation stack
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~5 seconds. Try 1 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~6 seconds. Try 1 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~6 seconds. Try 2 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~7 seconds. Try 2 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~4 seconds. Try 3 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~4 seconds. Try 3 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~4 seconds. Try 4 of 4
Recoverable error occurred (Invalid character in header content ["Authorization"]), sleeping for ~7 seconds. Try 4 of 4

✖ Stack x-voice-prod failed to deploy (21s)
Environment: darwin, node 14.19.3, framework 3.19.0, plugin 6.2.2, SDK 4.3.2
Credentials: Local, environment variables
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Invalid character in header content ["Authorization"]
```

### Environment information

```shell
Framework Core: 3.19.0
Plugin: 6.2.2
SDK: 4.3.2
```

**Comment by pgrzesik** [comment]
Hello 👋 Did you try deploying on a different machine or setting up the credentials once again from scratch? Does running regular `awscli` commands work for you for the same set of credentials?

**Comment by alexanderxc** [comment]
After some considerable trials it turns out that the issue was the online permission I set for the user I was deploying with. 

Serverless works perfectly with my account (which is an admin) but failed with a more restricted (although lenient) user account. 

I'm afraid that the permission scheme I found online had some bug in the "resources" description somewhere, although it is a bit difficult to correctly compose a correct permission scheme on AWS.

It would be incredibly helpful to have a section of the guide which better explains the required permission minimal and optimal for serverless deployment (considering some scenarios: only lambda vs. s3). 

Also it would be an immense improvement to receive a printed output of the schema exception that is thrown during deploy and not the general error reported above. These would have saved me hours of trial. 

This issue can be closed but there is room for improvement on the docs. 

Thanks for the great work anyway!
